connection_test.go

```diff
@@ -486,15 +508,70 @@ func TestPrivateLink(t *testing.T) {
 	}); err != nil {
 		t.Error(err)
 	}
+	defer func() {
+		os.Unsetenv(cacheServerURLEnv)
+		os.Unsetenv(ocspRetryURLEnv)
+	}()
+
 	ocspURL := os.Getenv(cacheServerURLEnv)
-	expectedURL := "http://ocsp.testaccount.us-east-1.privatelink.snowflakecomputing.com/ocsp_response_cache.json"
-	if ocspURL != expectedURL {
-		t.Errorf("expected: %v, got: %v", expectedURL, ocspURL)
-	}
+	assertEqualE(t, ocspURL, "http://ocsp.testaccount.us-east-1.privatelink.snowflakecomputing.com/ocsp_response_cache.json")
 	retryURL := os.Getenv(ocspRetryURLEnv)
-	expectedURL = "http://ocsp.testaccount.us-east-1.privatelink.snowflakecomputing.com/retry/%v/%v"
-	if retryURL != expectedURL {
-		t.Errorf("expected: %v, got: %v", expectedURL, retryURL)
+	assertEqualE(t, retryURL, "http://ocsp.testaccount.us-east-1.privatelink.snowflakecomputing.com/retry/%v/%v")
+}
+
+func TestOcspEnvVarsSetup(t *testing.T) {
+	ctx := context.Background()
+	for _, tc := range []struct {
+		host                string
+		cacheURL            string
+		privateLinkRetryURL string
+	}{
+		{
+			host:                "testaccount.us-east-1.snowflakecomputing.com",
+			cacheURL:            "", // no privatelink, default ocsp cache URL, no need to setup env vars
+			privateLinkRetryURL: "",
+		},
+		{
+			host:                "testaccount-no-privatelink.snowflakecomputing.com",
+			cacheURL:            "", // no privatelink, default ocsp cache URL, no need to setup env vars
+			privateLinkRetryURL: "",
+		},
+		{
+			host:                "testaccount.us-east-1.privatelink.snowflakecomputing.com",
+			cacheURL:            "http://ocsp.testaccount.us-east-1.privatelink.snowflakecomputing.com/ocsp_response_cache.json",
+			privateLinkRetryURL: "http://ocsp.testaccount.us-east-1.privatelink.snowflakecomputing.com/retry/%v/%v",
+		},
+		{
+			host:                "testaccount.cn-region.snowflakecomputing.cn",
+			cacheURL:            "http://ocsp.testaccount.cn-region.snowflakecomputing.cn/ocsp_response_cache.json",
+			privateLinkRetryURL: "", // not a privatelink env, no need to setup retry URL
+		},
+		{
+			host:                "testaccount.cn-region.privaTELINk.snowflakecomputing.cn",
+			cacheURL:            "http://ocsp.testaccount.cn-region.privatelink.snowflakecomputing.cn/ocsp_response_cache.json",
+			privateLinkRetryURL: "http://ocsp.testaccount.cn-region.privatelink.snowflakecomputing.cn/retry/%v/%v",
+		},
+		{
+			host:                "testaccount.some-region.privatelink.snowflakecomputing.mil",
+			cacheURL:            "http://ocsp.testaccount.some-region.privatelink.snowflakecomputing.mil/ocsp_response_cache.json",
+			privateLinkRetryURL: "http://ocsp.testaccount.some-region.privatelink.snowflakecomputing.mil/retry/%v/%v",
+		},
+	} {
+		t.Run(tc.host, func(t *testing.T) {
+			if err := setupOCSPEnvVars(ctx, tc.host); err != nil {
+				t.Errorf("error during OCSP env vars setup; %v", err)
+			}
+			defer func() {
+				os.Unsetenv(cacheServerURLEnv)
```

Comment:
Should we unset or reset to previous value?

Reply:
IMO unsetting it is enough, since setting this up is required when creating connection.

Reply:
So just to be on the safe side - if there is another test:
1. if it reuses a connection from pool - it's fine, because OCSP is already properly set up
2. if it creates a new connection/pool - it's fine, because it sets up this property again
Right?

Reply:
Exactly.